Description: videodecoder: allocate DMA DRM buffers with the DRM video info
 gst_video_decoder_decide_allocation_default did not consider the buffer
 being a DMA DRM buffer and got the wrong vinfo.
Author: Serge Schneider <serge@raspberrypi.com>
---
--- a/gst-libs/gst/video/gstvideodecoder.c
+++ b/gst-libs/gst/video/gstvideodecoder.c
@@ -4230,8 +4230,15 @@ gst_video_decoder_decide_allocation_defa
 
   gst_query_parse_allocation (query, &outcaps, NULL);
   gst_video_info_init (&vinfo);
-  if (outcaps)
-    gst_video_info_from_caps (&vinfo, outcaps);
+  if (outcaps) {
+    if (gst_video_is_dma_drm_caps (outcaps)) {
+      GstVideoInfoDmaDrm drm_info;
+      if (gst_video_info_dma_drm_from_caps (&drm_info, outcaps))
+        vinfo = drm_info.vinfo;
+    } else {
+      gst_video_info_from_caps (&vinfo, outcaps);
+    }
+  }
 
   /* we got configuration from our peer or the decide_allocation method,
    * parse them */
